sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -445,10 +445,17 @@ void peekMessages(MessagingEntityType entityType, boolean isSessionEnabled) {
 
         setReceiver(entityType, TestUtils.USE_CASE_DEFAULT, isSessionEnabled);
         // Act
-        final IterableStream<ServiceBusReceivedMessage> messages = receiver.peekMessages(maxMessages);
+        IterableStream<ServiceBusReceivedMessage> messages = receiver.peekMessages(maxMessages);
+        int actualCount = (int) messages.stream().count();
+
+        // 'peekMessages' API returns maxMessages but it can be less as well by design.
```

Comment:
Is this how the service is designed? If maxMessage is 10 and if the queue has > 10 messages, will the service ever return < 10 messages when `peek` is called?